entrypoint.sh

```diff
@@ -7,227 +7,247 @@ set -o pipefail
-  else
-    echo "The provided LOG_LEVEL of ${INPUT_LOG_LEVEL:-null or unset} is not valid"
-  fi
-
-  linter_failed="false"
-  declare -a linter_failures
-  declare -a linter_successes
+	# Set the preferred shell behavior
+	shopt -s globstar
+
+	# Set the default branch
+	export DEFAULT_BRANCH="main"
+
+	# Set workspace to /goat/ for local runs
+	export DEFAULT_WORKSPACE="/goat"
+
+	# Create variables for the various dictionary file paths
+	export GLOBAL_DICTIONARY="/etc/opt/goat/seiso_global_dictionary.txt"
+	export REPO_DICTIONARY="${GITHUB_WORKSPACE:-/goat}/.github/etc/dictionary.txt"
+
+	# Map certain environment variables
+	if [[ ${INPUT_DISABLE_TERRASCAN-} == "true" ]]; then
+		export VALIDATE_TERRAFORM_TERRASCAN="false"
+	fi
+
+	if [[ ${INPUT_DISABLE_MYPY-} == "true" ]]; then
+		export VALIDATE_PYTHON_MYPY="false"
+	fi
+
+	if [[ -n ${INPUT_EXCLUDE:+x} ]]; then
+		export FILTER_REGEX_EXCLUDE="${INPUT_EXCLUDE}"
+	fi
+
+	if [[ ${INPUT_LOG_LEVEL:='VERBOSE'} =~ ^(ERROR|WARN|NOTICE|VERBOSE|DEBUG|TRACE)$ ]]; then
+		export LOG_LEVEL="${INPUT_LOG_LEVEL}"
+		export ACTIONS_RUNNER_DEBUG="true"
+	fi
+
+	linter_failed="false"
+	declare -a linter_failures
+	declare -a linter_successes
 }
 
 function check_environment() {
-  # Check the GITHUB_BASE_REF (PRs only)
-  if [[ "${GITHUB_ACTIONS:-false}" == "true" && -n ${GITHUB_BASE_REF:+x} ]]; then
-    mainline="${GITHUB_BASE_REF##*/}"
-    if [[ "${mainline}" != "main" ]]; then
-      feedback ERROR "Base branch name is not main"
-    fi
-  fi
-
-  # Ensure dictionaries don't have overlap
-  overlap=$(comm -12 <(sort "${GLOBAL_DICTIONARY}" | tr '[:upper:]' '[:lower:]') \
-                     <(sort "${REPO_DICTIONARY}"   | tr '[:upper:]' '[:lower:]'))
-  if [[ "${overlap}" ]]; then
-    feedback WARNING "The following words are already in the global dictionary:
+	# Check the GITHUB_BASE_REF (PRs only)
+	if [[ ${GITHUB_ACTIONS:-false} == "true" && -n ${GITHUB_BASE_REF:+x} ]]; then
+		mainline="${GITHUB_BASE_REF##*/}"
+		if [[ ${mainline} != "main" ]]; then
+			feedback ERROR "Base branch name is not main"
+		fi
+	fi
+
+	# Ensure dictionaries don't have overlap
+	overlap=$(comm -12 <(sort "${GLOBAL_DICTIONARY}" | tr '[:upper:]' '[:lower:]') \
+		<(sort "${REPO_DICTIONARY}" | tr '[:upper:]' '[:lower:]'))
+	if [[ "${overlap}" ]]; then
+		feedback WARNING "The following words are already in the global dictionary:
 ${overlap}"
-    feedback ERROR "Overlap was detected in the per-repo and global dictionaries"
-  fi
+		feedback ERROR "Overlap was detected in the per-repo and global dictionaries"
+	fi
 }
 
-function super_lint() {
-  superlinter_logfile="/opt/goat/log/super-linter.log"
+function detect_kubernetes_file() {
+	local file="$1"
 
-  echo -e "\nRunning Super-Linter\n--------------------------\n"
-  echo "===============================" >> "$superlinter_logfile"
-  echo "SUPER-LINTER" >> "$superlinter_logfile"
+	if grep -q -v 'kustomize.config.k8s.io' "${file}" &&
```

Comment:
Can you please add a comment to explain why we want this?

Reply:
Done